include/api_manager/request.h

```diff
@@ -71,6 +71,9 @@ class Request {
   // Sets auth token to the request object. Caller of RequestHandler::Check
   // need to use it compose error message if authentication fails.
   virtual void SetAuthToken(const std::string &auth_token) = 0;
+  // Returns auth token. Used to compose authz_cache key in
+  // check_security_rules.cc
+  virtual std::string GetAuthToken() = 0;
```

Comment:
The different is: the former the token is stored in Nginx rgx_ctx.   the other the data is stored in request_context inside api_manager.   I prefer the latter.